File: leetcode/lc-us/cpp/2485.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int pivotInteger(int n) {
        int x = (n * n + n) >> 1;
        int st = (int) sqrt(x);
        if (st * st == x) return st;
        return -1;
    }
};

File: leetcode/lc-us/rust/2485.rs
impl Solution {
    pub fn pivot_integer(n: i32) -> i32 {
        let x = n * (n + 1) / 2;
        let y = (x as f64).sqrt() as i32;
        if y * y == x {
            return y as i32;
        }
        -1
    }
}